### PR TITLE
fix for #52 : solve parser error on numbers with leading +

### PR DESCRIPTION
**Pull request by WeinaJi**

### CODE REVIEW

**Comment by WeinaJi** [comment]
please retest

**Comment by ohm314** [comment]
please retest

**Comment by WeinaJi** [comment]
please retest

**Review by ohm314 — APPROVED** [review]
lgtm

File: src/parser/nmodl.yy [hunk 1]
```diff
@@ -707,6 +707,10 @@ number          :   NUMBER
                         $2->negate();
                         $$ = $2;
                     }
+                |   "+" NUMBER
```
Comment (pramodk) [inline src/parser/nmodl.yy:710]: @nrnhines : out of curiosity : for any reason this rule wasn't there in original implementation?

**Review by pramodk — CHANGES_REQUESTED** [review]
LGTM.
@WeinaJi : we should one test that checks this rule. See below file where we add nmodl constructs for testing : https://github.com/BlueBrain/nmodl/blob/6d39a657b8908933875f2f09b7dede3a9748757b/test/unit/utils/nmodl_constructs.cpp#L189

**Comment by WeinaJi** [comment]
> LGTM.
> @WeinaJi : we should one test that checks this rule. See below file where we add nmodl constructs for testing :
> 
> https://github.com/BlueBrain/nmodl/blob/6d39a657b8908933875f2f09b7dede3a9748757b/test/unit/utils/nmodl_constructs.cpp#L189

@pramodk hmm... there is one issue with test case ```testvisitor/Scenario: Convert AST back to NMODL form```
The "+" sign is removed after passing so when converting back to NMODL, the generated text doesn't match the origin format. I think that's expected, but how can I solve the matching error ?

**Comment by pramodk** [comment]
Sorry, missed your comment.

In that case you can pass an additional argument with the expected nmodl text. See e.g. https://github.com/BlueBrain/nmodl/blob/6d39a657b8908933875f2f09b7dede3a9748757b/test/unit/utils/nmodl_constructs.cpp#L396

**Review by pramodk — APPROVED** [review]

**Comment by pramodk** [comment]
thanks @WeinaJi !

**Pull request merged**